### PR TITLE
[FrameworkBundle] Deprecate the "--env" and "--no-debug" console options

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | n/a
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

See https://github.com/symfony/symfony/issues/23343#issuecomment-345609612

### CODE REVIEW

**Review by dunglas — APPROVED** [review]

**Review by fabpot — CHANGES_REQUESTED** [review]
You should add a note in the UPGRADE/CHANGELOG files

**Review by OskarStark — APPROVED** [review]

**Comment by chalasr** [comment]
@fabpot done, thanks for the reminder.
Also moved to FrameworkBundle, I've been confused by the console recipe relying on them.

File: UPGRADE-4.2.md [hunk 1]
```diff
@@ -110,6 +110,10 @@ FrameworkBundle
    ```
  * The `ContainerAwareCommand` class has been deprecated, use `Symfony\Component\Console\Command\Command`
    with dependency injection instead.
+ * The `--env` console option and its "-e" shortcut have been deprecated,
+   set the "APP_ENV" environment variable instead.
+ * The `--env` console option and its "-e" shortcut have been deprecated,
+   set the "APP_DEBUG" environment variable to "0" instead.
```
Comment (jvasseur) [inline UPGRADE-4.2.md]: `-no-debug` instead of `--env` :wink:

Comment (chalasr) [reply UPGRADE-4.2.md]: fixed thanks

File: UPGRADE-5.0.md [hunk 2]
```diff
@@ -116,6 +116,10 @@ FrameworkBundle
  * Added support for the SameSite attribute for session cookies. It is highly recommended to set this setting (`framework.session.cookie_samesite`) to `lax` for increased security against CSRF attacks.
  * The `ContainerAwareCommand` class has been removed, use `Symfony\Component\Console\Command\Command`
    with dependency injection instead.
+ * The `--env` console option and its "-e" shortcut have been deprecated,
```
Comment (Koc) [inline UPGRADE-5.0.md]: In 5.0 this option would removed.

Comment (chalasr) [reply UPGRADE-5.0.md]: good catch, fixed

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Comment by weaverryan** [comment]
Can you open a docs issue / or?

**Comment by chalasr** [comment]
@weaverryan sorry, forgot about it. Just opened https://github.com/symfony/symfony-docs/issues/10444

**Comment by ogizanagi** [comment]
@gnugat : This was discussed a bit a long time ago. Here is a Fabien's https://github.com/symfony/symfony/issues/23343#issuecomment-345609612 mentioning this.
Note there is no BC break here, as it's still supported until 5.0. It'll be considered a BC break when removing it at time `master` becomes the v5 branch. For now, this is only a deprecation.

**Comment by chalasr** [comment]
@gnugat link added to the PR body, sorry about that

**Comment by c33s** [comment]
this is really a bad decision, it makes the use on windows much more complicated. this is really a very bad DX.

```
php bin/console --env=dev
Symfony 3.4.12 (kernel: src, env: dev, debug: true)
(fully working symfony console with no errors)
```
vs
```
APP_ENV=dev php bin\console
'APP_ENV' is not recognized as an internal or external command,
operable program or batch file.
```
vs
```
λ cmd /V /C "set APP_ENV=dev&& php bin\console"
 [WARNING] Some commands could not be registered:
In EnvVarProcessor.php line 76:
  Environment variable not found: "DATABASE_URL_LEGACY".
Symfony 3.4.12 (kernel: src, env: dev, debug: true)
---snip---
In EnvVarProcessor.php line 76:
  Environment variable not found: "APP_SECRET".
```
also the handy autocomplete possibility with clink on windows and bash-completion on linux is gone with this change. environment variables are cool but not the holy grail for everything. there is a reason for having flags on commands. imagine you have to call all commands like this:

```
URL=http://example.com RECURSIVE=true OUTPUT_FILE=foo.html wget
```
please revert this.

**Comment by nicolas-grekas** [comment]
@c33s can you please open a separate issue? Commenting on anything closed is likely going to get low traction.

**Comment by nicolas-grekas** [comment]
Reverted in #29204

**Pull request merged**